contracts/interchain-token-service/src/contract/execute.rs

```diff
@@ -196,6 +197,49 @@ pub fn set_chain_config(
     }
 }
 
+#[allow(unused)]
+fn translate_and_save_token_decimals_on_token_deployment(
+    storage: &mut dyn Storage,
+    src_chain: &ChainNameRaw,
+    dest_chain: &ChainNameRaw,
+    message: Message,
+) -> Result<Message, Error> {
+    match message {
+        Message::DeployInterchainToken {
+            token_id,
+            name,
+            symbol,
+            decimals: src_chain_decimals,
+            minter,
+        } => {
+            let src_chain_config =
+                state::load_chain_config(storage, src_chain).change_context(Error::State)?;
+            let dest_chain_config =
+                state::load_chain_config(storage, dest_chain).change_context(Error::State)?;
+
+            let dest_chain_decimals = if src_chain_config.max_uint.le(&dest_chain_config.max_uint) {
+                src_chain_decimals
+            } else {
+                src_chain_config.max_target_decimals
```

Comment:
do we want to check if this amount of decimals can actually fit in the max_uint of the destination chain?

Reply:
How do we check without knowing what a reasonable amount is coming from the source chain? Checking it for `Uint256:MAX` would fail with even zero decimals against u64.